orchestrator/crates/prover-clients/atlantic-service/src/client.rs

```diff
@@ -62,13 +67,102 @@ impl AtlanticClient {
         Self { client, proving_layer }
     }
 
+    pub async fn get_artifacts(&self, artifact_path: String) -> Result<Vec<u8>, AtlanticError> {
+        debug!("Getting artifacts from {}", artifact_path);
+        let client = reqwest::Client::new();
+        let response = client.get(&artifact_path).send().await.map_err(AtlanticError::GetJobResultFailure)?;
+
+        if response.status().is_success() {
+            let response_text = response.bytes().await.map_err(AtlanticError::GetJobResultFailure)?;
+            Ok(response_text.to_vec())
+        } else {
+            Err(AtlanticError::AtlanticService(response.status()))
+        }
+    }
+
+    pub async fn get_bucket(&self, bucket_id: &str) -> Result<AtlanticGetBucketResponse, AtlanticError> {
+        let response =
+            self.client.request().method(Method::GET).path("buckets").path(bucket_id).send().await.map_err(|e| {
+                tracing::error!("Failed to get bucket status, {}", e);
+                AtlanticError::GetBucketStatusFailure(e)
+            })?;
+
+        match response.status().is_success() {
+            true => response.json().await.map_err(|e| {
+                tracing::error!("Failed to parse bucket status, {}", e);
+                AtlanticError::GetBucketStatusFailure(e)
+            }),
+            false => Err(AtlanticError::AtlanticService(response.status())),
+        }
+    }
+
+    pub async fn create_bucket(
+        &self,
+        atlantic_api_key: impl AsRef<str>,
+    ) -> Result<AtlanticBucketResponse, AtlanticError> {
+        let response = self
+            .client
+            .request()
+            .method(Method::POST)
+            .header(ACCEPT, HeaderValue::from_static("application/json"))
+            .header(CONTENT_TYPE, HeaderValue::from_static("application/json"))
+            .path("buckets")
+            .query_param("apiKey", atlantic_api_key.as_ref())
+            .body(AtlanticCreateBucketRequest {
+                external_id: None,
+                node_width: None,
+                aggregator_version: AtlanticAggregatorVersion::SnosAggregator0_13_3,
+                aggregator_params: AtlanticAggregatorParams { use_kzg_da: true, full_output: false },
+            })
+            .map_err(AtlanticError::CreateBucketFailure)?
+            .send()
+            .await
+            .map_err(AtlanticError::AddJobFailure)?;
```

Comment:
something like this would be more helpful!
```suggestion
            .map_err(AtlanticError::AddJobFailure("Bucket"))?;
```

Reply:
Update the error type